provider/internal/java/filter.go

```diff
@@ -191,10 +183,61 @@ func (p *javaServiceClient) convertSymbolRefToIncidentContext(symbol protocol.Wo
 			Character: ref.Range.End.Character,
 		},
 	}
-	incident.Variables[FILE_KEY] = ref.URI
 	lineNumber := int(ref.Range.Start.Line)
 	incident.LineNumber = &lineNumber
 
 	return incident, nil
 
 }
+
+func (p *javaServiceClient) getURI(refURI string) (uri.URI, error) {
+	if !strings.HasPrefix(refURI, FILE_URI_PREFIX) {
+		return uri.Parse(refURI)
+	}
+
+	u, err := url.Parse(refURI)
+	if err != nil {
+		return uri.URI(""), err
+	}
+
+	// Decompile the jar
+	sourceRange, err := strconv.ParseBool(u.Query().Get("source-range"))
+	if err != nil {
+		// then we got some response that does not make sense or should not be valid
+		return uri.URI(""), fmt.Errorf("unable to get konveyor-jdt source range query parameter")
+	}
+	packageName := u.Query().Get("packageName")
+
+	var jarPath string
+	if sourceRange {
+		// If there is a source range, we know we know there is a sources jar
+		jarName := filepath.Base(u.Path)
+		s := strings.TrimSuffix(jarName, ".jar")
+		s = fmt.Sprintf("%v-sources.jar", s)
+		jarPath = filepath.Join(filepath.Dir(u.Path), s)
+	} else {
```

Comment:
interesting, so we did find the solution for non source?

Reply:
I don't know if this is working as intended and/or fixes the full problem. It does solve the other big problem that folks are bringing up, so it is worth getting in, and then I will add some follow up after testing tomorrow to see how off it all is.